### PR TITLE
Add support for NuSpecs

### PR DESCRIPTION
**Pull request by ericstj**
This brings over the NuGet.Core code that existed for
loading NuSpecs.  I also added the File list to the
manifest to support builing a package from a
nuspec, and the ability to save a nuspec.

In most cases I tried to preserve the NuGet.Core
code, though I did drop "exclude empty directories"
since this functionality was very confusing in nuget.exe.
(you had to specify -ExcludeEmptyDirectories in order
for it to keep the empty directories designated by
placeholder files)

/cc @davidfowl @emgarten

### CODE REVIEW

File: src/Microsoft.DotNet.Tools.Pack/NuGet/PathResolver.cs [hunk 1]
```diff
@@ -0,0 +1,301 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace NuGet
+{
+    public static class PathResolver
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Pack/NuGet/PathResolver.cs]: I'm not sure we want to keep this code in the long run. We're using a new globbing library that came from ASP.NET 5 for other scenarios but for compat I guess it's fine.

Comment (ericstj) [inline src/Microsoft.DotNet.Tools.Pack/NuGet/PathResolver.cs]: At first I had left it out entirely and thrown on any wildcard in the path.  We don't need it for our scenarios.  If you'd prefer I can do that here.

**Comment by emgarten** [comment]
Which branch/repo was the NuGet.Core code taken from?

**Comment by ericstj** [comment]
https://github.com/NuGet/NuGet2/tree/dev/src/Core/Authoring.

Looks like the 2.11 branch may have some newer code in both the files I ported as well as those that David previously ported.

**Comment by emgarten** [comment]
Yes, 2.11 is the newest. It has support for contentFiles and include/exclude.

**Comment by ericstj** [comment]
Also seeing https://github.com/NuGet/NuGet2/blob/2.11/src/Core/Authoring/ManifestReader.cs that might be a better solution than https://github.com/dotnet/cli/pull/800/files#diff-162ff40444ca50ec5dfda2fa5f26c1daR77.  Let me take a look at a diff with 2.11 and see if I can reconcile that.

**Comment by ericstj** [comment]
Ok, I brought over those features from 2.11.  I also cleaned up the diff a bit and ported ManifestReader instead of the rewrite I did in PackageMetadataXmlExtensions.  I diffed my impl and ManifestReader (round tripping a nuspec) and noticed only a couple string.trim related diffs.

**Comment by ericstj** [comment]
@dotnet-bot please test Ubuntu Release Build

**Comment by ericstj** [comment]
@emgarten can you please have a look at the latest?

**Comment by emgarten** [comment]
This looks good :shipit:

**Comment by bricelam** [comment]
My hero! (resolves aspnet/dnx#45)

**Comment by ericstj** [comment]
This still isn't plumbed through the command.  Right now I'm just doing it for source sharing.  We also need to stand up some basic commandline functionality.  If their is interest in that I can mock something up in another PR.

**Comment by bricelam** [comment]
Ah, I see. We've been able to workaround most limitations thus far, so it can wait from my team's perspective.

**Pull request merged**